### DISCUSSION TITLE
Events API accepting requests without Bearer Token

### DISCUSSION THREAD

**Discussion by chandru89new** [plausible/analytics#1362] [Anything Else]
I am noticing that the events API accepts requests without authorization. It also updates the data for the domain. Shouldn't this API too be authorized like the Stats API?

**Comment by ukutaht** [comment]
This is by design.

The web is a public platform and any authentication token you would use could be inspected by any visitor. Since the token needs to be public in order to send events from your website, it doesn't really work as any sort of security mechanism. The Events API is designed as a public API with no authentication requirements.

**Reply by chandru89new** [reply]
Hmm, I think I get your point. Thanks!

**Reply by ragamuf** [reply]
Not sure I agree with this comment w.r.t making the call from the server side. I ought to have the option to secure it from there with a token that only resides on the server. Thoughts?

**Reply by ukutaht** [reply]
I guess it would be kinda nice if you're only sending server-to-server.

The other consideration is that accepting pageviews must be cheap. At peak load we're ingesting over 1k requests per second. For proper security you have to fetch the relevant API key from the database on every API request but that would be by far the most expensive part of our ingestion pipeline. We don't do any network request when ingesting data because it just takes too much time.

It could work if we cache the API key for some amount of hours but even then it would be a significant burden.

I'm curious, do you have examples of other analytics platforms that allow you to send pageviews using a token? I'm not aware of any.

**Reply by ragamuf** [reply]
Appreciate the feedback. I will have to look at what we use at work more closely but I assume so as it is a mobile app in this case. And I do understand the trade offs you are making here.
